src/js/dp/feedback-form.js

```diff
@@ -0,0 +1,122 @@
+
+            const formErrors = document.querySelectorAll(
+                "#feedback-form-page-container .feedback-form-error"
+            );
+            formErrors.forEach((formError) => {
+                formError.remove();
+            });
+
+            const emailField = document.querySelector("#email-field");
+            const descriptionField = document.querySelector("#description-field");
+            let hasErrors = false;
+
+            if (descriptionField && descriptionField.value === "") {
+                const descriptionError =
+                    '<span class="feedback-form-error" role="alert">Write some feedback</span>';
+                if (!document.querySelector("#description-field-label .feedback-form-error")) {
+                    const descriptionFieldLabel = document.querySelector(
+                        "#description-field-label"
+                    );
+                    descriptionFieldLabel.insertAdjacentHTML(
+                        "afterend",
+                        descriptionError
+                    );
+                }
+                descriptionField.classList.add("feedback-form-control__error");
+                descriptionField.focus();
+                hasErrors = true;
+            }
+
+            if (emailField && emailField.value !== "") {
+                let emailError = "";
+                if (hasErrors) {
+                    emailError =
+                        '<span class="feedback-form-error" role="alert" aria-live="polite">This is not a valid email address, correct it or delete it</span>';
+                } else {
+                    emailError =
+                        '<span class="feedback-form-error" role="alert">This is not a valid email address, correct it or delete it</span>';
+                }
+
+                const emailRegex = /^[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,6}$/g;
+                if (!emailRegex.test(emailField.value)) {
+                    if (!document.querySelector("#email-field-label .feedback-form-error")) {
+                        const emailFieldLabel =
+                            document.querySelector("#email-field-label");
+                        emailFieldLabel.insertAdjacentHTML("beforeend", emailError);
+                    }
+                    emailField.classList.add("feedback-form-control__error");
+                    hasErrors = true;
+                }
+            }
+
+            if (hasErrors) {
+                const feedbackButton = document.querySelector(".feedback-btn")
+                feedbackButton.blur();
+                return;
+            }
+
+            const { request, serializedData } = initFeedbackRequestHandler(feedbackFormContainer, feedbackURL);
+
+            request.send(serializedData);
+        });
+    }
+});
+
+function initFeedbackRequestHandler(form, path, feedbackMessageError) {
+    const serializedData = serializeFormData(form);
+    const request = new XMLHttpRequest();
```

Comment:
Think we can migrate to the [fetch api](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch)